pandas/tests/io/json/conftest.py

```diff
@@ -7,3 +7,27 @@ def orient(request):
     Fixture for orients excluding the table format.
     """
     return request.param
+
+
+@pytest.fixture
+def json_dir_path(datapath):
+    """
+    The directory path to the data files needed for parser tests.
+    """
+    return datapath("io", "json", "data")
+
+
+@pytest.fixture(params=["ujson", "pyarrow"])
+def engine(request):
+    return request.param
+
+
+@pytest.fixture
+def json_engine_pyarrow_xfail(request):
+    """
+    Fixture that xfails a test if the engine is pyarrow.
+    """
+    engine = request.getfixturevalue("engine")
+    if engine == "pyarrow":
+        mark = pytest.mark.xfail(reason="pyarrow doesn't support this.")
```

Comment:
Yeah I would prefer this to be added to _each_ xfailed test so the `reason` more clearly describes the exact thing that is not supported

Reply:
Got it, I think I added a comment on each xfail but this makes more sense.

Reply:
```
def test_foo(request, ...)
    if engine == "pyarrow":
        request.node.add_marker(pytest.mark.xfail(reason="the comment"))
```